File: SampleOCPod/SampleOCPod/ViewController.m
//
//  ViewController.m
//  SampleOCPod
//
//  Created by 68 on 2024/10/10.
//

#import "ViewController.h"

@interface ViewController () <UITableViewDataSource, UITableViewDelegate>
@property (nonatomic, strong) UITableView *tableView;
@property (nonatomic, strong) NSArray<NSDictionary<NSString *, Class> *> *dataList;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.dataList = @[
        @{@"title": @"授权demo", @"clz": NSClassFromString(@"AuthDemoVC")},
        @{@"title": @"68号加好友demo", @"clz": NSClassFromString(@"P2PDemoVC")},
        @{@"title": @"群分享链接进入群聊demo", @"clz": NSClassFromString(@"GroupShareLinkDemoVC")},
        @{@"title": @"群别名进入群聊demo", @"clz": NSClassFromString(@"GroupAlianNameDemoVC")},
        @{@"title": @"otc展示demo", @"clz": NSClassFromString(@"OTCDemoVC")},
        @{@"title": @"帮h5获取AccessToken", @"clz": NSClassFromString(@"AccessTokenVC")},
    ];
    
    self.tableView = [[UITableView alloc] initWithFrame:self.view.bounds style:UITableViewStyleGrouped];
    [self.tableView registerClass: [UITableViewCell class] forCellReuseIdentifier: @"cell"];
    self.tableView.dataSource = self;
    self.tableView.delegate = self;
    [self.view addSubview: self.tableView];
    [self.tableView reloadData];
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataList.count;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return 60;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier: @"cell" forIndexPath:indexPath];
    cell.backgroundColor = UIColor.whiteColor;
    NSDictionary *data = self.dataList[indexPath.row];
    cell.textLabel.text = data[@"title"];
    cell.textLabel.textColor = UIColor.blackColor;
    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    NSDictionary *data = self.dataList[indexPath.row];
    Class vcClz = data[@"clz"];
    UIViewController *vc = [[vcClz alloc] init];
    [self.navigationController pushViewController:vc animated:YES];
}
@end

File: Example/Example/ViewController.swift
//
//  ViewController.swift
//  TestOSIM
//
//  Created by flow on 9/20/24.
//

import UIKit

class ViewController: UIViewController, UITableViewDataSource,UITableViewDelegate {

    let vcList: [(String, BaseDemoVC.Type)] = [
        ("授权demo", AuthDemoVC.self),
        ("68号加好友demo", P2PDemoVC.self),
        ("群分享链接进入群聊demo", GroupShareLinkDemoVC.self),
        ("群别名进入群聊demo", GroupAlianNameDemoVC.self),
        ("otc展示demo", OTCDemoVC.self),
        ("综合demo", FullDemoVC.self),
    ]
    lazy var tableView: UITableView = {
        let v = UITableView(frame: UIScreen.main.bounds, style: .grouped)
        v.delegate = self
        v.dataSource = self
        v.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
        v.backgroundColor = .white
        return v
    }()
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = .white
        self.view.addSubview(tableView)
        self.title = "OSIM例子"
    }
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return vcList.count
    }
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 60
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
        cell.backgroundColor = .white
        cell.textLabel?.text = vcList[indexPath.row].0
        cell.textLabel?.textColor = .black
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        let config = vcList[indexPath.row]
        let vcClz = config.1
        let vc: BaseDemoVC = vcClz.init()
        vc.topLabel.text = config.0
        self.navigationController?.pushViewController(vc, animated: true)
    }
}

